server/service/osquery_utils/queries.go

```diff
@@ -1303,6 +1310,29 @@ func directIngestDiskEncryption(ctx context.Context, logger log.Logger, host *fl
 	return ds.SetOrUpdateHostDisksEncryption(ctx, host.ID, encrypted)
 }
 
+func directIngestDiskEncryptionKey(ctx context.Context, logger log.Logger, host *fleet.Host, ds fleet.Datastore, rows []map[string]string, failed bool) error {
+	if len(rows) == 0 || failed {
+		// assume the extension is not there
+		return nil
+	}
+
+	var key string
+	for i, row := range rows {
+		if strings.Contains(row["line"], "RecoveryKey") {
+			keyReader := strings.NewReader(rows[i+1]["line"])
```

Comment:
This could panic if "RecoveryKey" is found on the last row in `rows`. You should also check in the `if` condition above that `i < len(rows)-1`.

Reply:
@mna in general about this and all the other comments: maybe we should build and parse the full XML object? what do you think?

Reply:
Yes, that's a strong +1 for me! You only need the field you care about in the target struct (`RecoveryKey`).